### ISSUE TITLE
Warnings (soon-to-be) emitted in the Core multiprocess build

### ISSUE THREAD

**Issue by fanquake** [chaincodelabs/libmultiprocess#77]
We are looking at [re-enabling a more comprehensive](https://github.com/bitcoin/bitcoin/pull/25972) set of warnings (by default) for depends builds, and currently [the multiprocess CI job](https://github.com/bitcoin/bitcoin/pull/25972/checks?check_run_id=8348618951) is emitting a few:

```bash
In file included from ipc/capnp/protocol.cpp:14:
/tmp/cirrus-ci-build/depends/i686-pc-linux-gnu/include/mp/proxy-io.h:33:37: error: parameter 'connection' shadows member inherited from type 'InvokeContext' [-Werror,-Wshadow-field]
    ClientInvokeContext(Connection& connection, ThreadContext& thread_context)
                                    ^
/tmp/cirrus-ci-build/depends/i686-pc-linux-gnu/include/mp/proxy-io.h:27:17: note: declared here
    Connection& connection;
                ^
/tmp/cirrus-ci-build/depends/i686-pc-linux-gnu/include/mp/proxy-io.h:160:16: error: moving a local object in a return statement prevents copy elision [-Werror,-Wpessimizing-move]
        return std::move(logger);
               ^
/tmp/cirrus-ci-build/depends/i686-pc-linux-gnu/include/mp/proxy-io.h:160:16: note: remove std::move call here
        return std::move(logger);
               ^~~~~~~~~~      ~
2 errors generated.
```

**Comment by ryanofsky** [comment]
Thanks, these should be pretty easy to fix. Will try to do it soon or can merge a PR.

**Comment by hebasto** [comment]
`-Wpessimizing-move` warnings have been already fixed in #58.
